components/styled/table/src/molecules/table-methods.tsx

```diff
@@ -100,4 +103,114 @@ const prependColumnWithExpansionControls = (
   return [expanderColumn].concat(columns)
 }
 
-export default createColumns
+const calculateStaticColumnOffset = (cellIdx: number, staticColumns: number, firstColumnWidth: number) => {
+  if (staticColumns === 0) {
+    return undefined
+  }
+
+  if (cellIdx === 0) {
+    return 0
+  }
+
+  if (cellIdx < staticColumns) {
+    return firstColumnWidth
+  }
+
+  return undefined
+}
+
+const handleScrollEvent = (table: Table<DataEntity>, headerAxis: Axis, scrollState: ScrollState) => {
+  const { scrollWidth, scrollHeight, currentXScroll, currentYScroll } = scrollState
+  if (
+    headerAxis === 'x' &&
+    currentYScroll > scrollHeight - 20 &&
+    currentYScroll < scrollHeight + 20 &&
+    table.getCanNextPage()
+  ) {
+    table.nextPage()
+  }
+
+  if (
+    headerAxis === 'y' &&
+    currentXScroll > scrollWidth - 20 &&
+    currentXScroll < scrollWidth + 20 &&
+    table.getCanNextPage()
+  ) {
+    table.nextPage()
+  }
+}
+
+const handleScrollEventManual = (
+  getCanNextPage: () => boolean,
+  nextPage: () => void,
+  headerAxis: Axis,
+  scrollState: ScrollState
+) => {
+  const { scrollWidth, scrollHeight, currentXScroll, currentYScroll } = scrollState
+  if (
+    headerAxis === 'x' &&
+    currentYScroll > scrollHeight - 20 &&
+    currentYScroll < scrollHeight + 20 &&
+    getCanNextPage()
+  ) {
+    nextPage()
+  }
+
+  if (
+    headerAxis === 'y' &&
+    currentXScroll > scrollWidth - 20 &&
+    currentXScroll < scrollWidth + 20 &&
+    getCanNextPage()
+  ) {
+    nextPage()
+  }
+}
+
+const handleDataUpdate = (
+  tableData: TableData,
+  pageIndex: number,
+  pageSize: number,
+  headerAxis: Axis,
+  setColumns: (value: React.SetStateAction<ColumnDef<DataEntity>[]>) => void,
+  setData: (value: React.SetStateAction<DataEntity[]>) => void,
+  setPageCount: (value: React.SetStateAction<number>) => void
+) => {
+  if (headerAxis === 'x') {
+    setColumns(createColumns(tableData))
+    setData(tableData.rows.slice(0, (pageIndex + 1) * pageSize))
+    setPageCount(Math.ceil(tableData.rows.length / pageSize))
+  } else {
+    const head = tableData.headers[0]
```

Comment:
You could get an index out of bounds exception here is the headers array is empty - same goes for the slice methods